File: java/src/main/java/com/smeltery/service/config/SslConfig.java
package com.smeltery.service.config;

import java.io.IOException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.io.Resource;

@Configuration
public class SslConfig {

    private static final Logger logger = LoggerFactory.getLogger(SslConfig.class);

    @Value("${server.ssl.trust-store}")
    private Resource trustStoreResource;

    @Value("${server.ssl.trust-store-password}")
    private String trustStorePassword;

    @Value("${server.ssl.key-store}")
    private Resource keyStoreResource;

    @Value("${server.ssl.key-store-password}")
    private String keyStorePassword;

    @Bean
    public void setUpTrustStoreForApplication() throws IOException {
        String trustStoreFilePath = trustStoreResource.getFile().getAbsolutePath();

        logger.info("TrustStore location is {}", trustStoreFilePath);

        System.setProperty("javax.net.ssl.trustStore", trustStoreFilePath);
        System.setProperty("javax.net.ssl.trustStorePassword", trustStorePassword);
    }

    @Bean
    public void setUpKeyStoreForApplication() throws IOException {
        String keyStoreFilePath = keyStoreResource.getFile().getAbsolutePath();

        logger.info("KeyStore location is {}", keyStoreFilePath);

		System.setProperty("javax.net.ssl.keyStore", keyStoreFilePath);
        System.setProperty("javax.net.ssl.keyStorePassword", keyStorePassword);
    }
}

File: kotlin/src/main/kotlin/com/ibm/service/config/SslConfig.kt
package com.smeltery.service.config

import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.io.Resource
import java.io.IOException

@Configuration
class SslConfig {
	@Value("\${server.ssl.trust-store}")
	private val trustStoreResource: Resource? = null

	@Value("\${server.ssl.trust-store-password}")
	private val trustStorePassword: String? = null

	@Value("\${server.ssl.key-store}")
	private val keyStoreResource: Resource? = null

	@Value("\${server.ssl.key-store-password}")
	private val keyStorePassword: String? = null

	@Bean
	@Throws(IOException::class)
	fun setUpTrustStoreForApplication() {
		val trustStoreFilePath = trustStoreResource!!.file.absolutePath
		logger.info("TrustStore location is {}", trustStoreFilePath)
		System.setProperty("javax.net.ssl.trustStore", trustStoreFilePath)
		System.setProperty("javax.net.ssl.trustStorePassword", trustStorePassword)
	}

	@Bean
	@Throws(IOException::class)
	fun setUpKeyStoreForApplication() {
		val keyStoreFilePath = keyStoreResource!!.file.absolutePath
		logger.info("KeyStore location is {}", keyStoreFilePath)
		System.setProperty("javax.net.ssl.keyStore", keyStoreFilePath)
		System.setProperty("javax.net.ssl.keyStorePassword", keyStorePassword)
	}

	companion object {
		private val logger = LoggerFactory.getLogger(SslConfig::class.java)
	}
}
